### PR TITLE
List the cherry-picked PRs in the generated PR that promotes an cherry-picked AMP version

### LINKED ISSUE #160: FR: Cherry Pick  promotion PRs should link to the CP Issue and relevant PRs
**summary**
AMP's old release issue trackers used to be a useful place to see how an RTV was deployed over time, and all of the CPs that were applied to it.

Ideally this new system could provide a similar view.
The CP Promote PRs should link to the relevant CP Issue and PRs. 

For example, https://github.com/ampproject/cdn-configuration/pull/159 should link to https://github.com/ampproject/amphtml/issues/37715 and https://github.com/ampproject/amphtml/pull/37671

### PR DESCRIPTION
**Pull request by danielrozenberg**
Fixes #160

Example of a PR that gets generated with this code: https://github.com/danielrozenberg/cdn-configuration/pull/42

I wrapped the new piece of code in a fail-safe try/catch so that it wouldn't fail the entire CP if anything in the API calls borks

// cc: @samouri

### CODE REVIEW

**Comment by danielrozenberg** [comment]
> Getting PRs won't work when there's a merge conflict. It would be better to list commit shas.

I feel like it's a rare enough situation that we don't have to worry about it too much, especially if ...

> Another option is to add a link to the branch, which would just be something like https://github.com/ampproject/amphtml/commits/amp-release-2202142035002

... I do this and add that regardless of whether getting the PR numbers succeeded or failed

**Comment by estherkim** [comment]
> rare enough situation

Still should support it. If a PR isn't found, can you add a link to the commit?

File: scripts/promote-cherry-picks.ts [hunk 1]
```diff
@@ -37,8 +45,52 @@ function getAmpVersionToCherrypick(
...
+  if (cherryPickedPRs.length) {
+    body += '\n\nPRs included in this cherry pick:\n';
+    body += cherryPickedPRs
+      .map(
+        (pullNumber) =>
+          `* https://github.com/ampproject/amphtml/pull/${pullNumber}`
+      )
+      .join('\n');
+  }
+  body += `\n\n[Release branch commits](https://github.com/ampproject/amphtml/commits/amp-release-${ampVersion})`;
```
Comment (estherkim) [inline scripts/promote-cherry-picks.ts:88]: Oops I missed this. Can the link say the branch name?

File: scripts/promote-cherry-picks.ts [hunk 2]
```diff
@@ -37,8 +45,52 @@ function getAmpVersionToCherrypick(
...
+      sha: ampVersion,
+      per_page: numberOfCherryPickedCommits,
+    });
+    return data
+      .map(({commit}) => {
+        const [firstLine] = commit.message.split('\n');
+        const matches = firstLine?.match(/\(#(?<pullNumber>\d+)\)$/);
+        return matches?.groups?.pullNumber;
+      })
+      .filter(defined);
```
Comment (estherkim) [inline scripts/promote-cherry-picks.ts:65]: Can the map function be changed to not returned undefined instead of filtering?

**Review by estherkim — COMMENTED** [review]
PR body looks awesome

File: scripts/promote-cherry-picks.ts (hunk 2 above)
Comment (danielrozenberg) [reply scripts/promote-cherry-picks.ts:65]: Well the mapping function has to return _something_ even if that something is `undefined`. We'd still have to filter out the items that didn't succeed in extracting the PR number

File: scripts/promote-cherry-picks.ts (hunk 1 above)
Comment (danielrozenberg) [reply scripts/promote-cherry-picks.ts]: You didn't miss this, I added it after your previous round of comments 😅
Done: 99f9b3d

File: scripts/promote-cherry-picks.ts [hunk 3]
```diff
@@ -37,8 +45,52 @@ function getAmpVersionToCherrypick(
...
+  channels: string,
+  cherryPickedPRs: string[]
+): string {
+  let body = `Promoting release ${ampVersion} to channels: ${channels}`;
+  if (cherryPickedPRs.length) {
+    body += '\n\nPRs included in this cherry pick:\n';
+    body += cherryPickedPRs
+      .map(
+        (pullNumber) =>
+          `* https://github.com/ampproject/amphtml/pull/${pullNumber}`
```
Comment (estherkim) [inline scripts/promote-cherry-picks.ts:84]: Might have gotten lost but can you add a link to the commit if it doesn't have a PR (this happens with merge conflicts)

File: scripts/promote-cherry-picks.ts (hunk 2 above)
Comment (estherkim) [reply scripts/promote-cherry-picks.ts:65]: Why not just use filter and get rid of map?

File: scripts/promote-cherry-picks.ts (hunk 3 above)
Comment (danielrozenberg) [reply scripts/promote-cherry-picks.ts]: Done 2190821

**Review by samouri — APPROVED** [review]

**Review by estherkim — APPROVED** [review]
Thank youuu

**Pull request merged**